docs/operators/csv_parser.md

```diff
@@ -0,0 +1,161 @@
+| `preserve_to` |                  | Preserves the unparsed value at the specified [field](/docs/types/field.md)                                                                                                                                                              |
+| `on_error`    | `send`           | The behavior of the operator if it encounters an error. See [on_error](/docs/types/on_error.md)                                                                                                                                          |
+| `timestamp`   | `nil`            | An optional [timestamp](/docs/types/timestamp.md) block which will parse a timestamp field before passing the entry to the output operator                                                                                               |
+| `severity`    | `nil`            | An optional [severity](/docs/types/severity.md) block which will parse a severity field before passing the entry to the output operator                                                                                                  |
+
+### Example Configurations
+
+#### Parse the field `message` with a csv parser
+
+Configuration:
+
+```yaml
+- type: csv_parser
+  parse_from: message
+  header: id,severity,message
+```
+
+<table>
+<tr><td> Input Entry </td> <td> Output Entry </td></tr>
+<tr>
+<td>
+
+```json
+{
+  "timestamp": "",
+  "resource": { },
+  "attributes": { },  
+  "body": {
+    "message": "1,debug,\"\"Debug Message\"\""
+  }
+}
+```
+
+</td>
+<td>
+
+```json
+{
+  "timestamp": "",
+  "resource": { },
+  "attributes": { },  
+  "body": {
+    "id": "1",
+    "severity": "debug",
+    "message": "\"Debug Message\""
+  }
+}
+```
+
+</td>
+</tr>
+</table>
+
+#### Parse the field `message` with a csv parser using tab delimiter
+
+Configuration:
+
+```yaml
+- type: csv_parser
+  parse_from: message
+  header: id,severity,message
+  delimiter: "\t"
+```
+
+<table>
+<tr><td> Input Entry </td> <td> Output Entry </td></tr>
+<tr>
+<td>
+
+```json
+{
+  "timestamp": "",
+  "resource": { },
+  "attributes": { },  
+  "body": {
+    "message": "1 debug \"Debug Message\""
+  }
+}
+```
+
+</td>
+<td>
+
+```json
+{
+  "timestamp": "",
+  "resource": { },
+  "attributes": { },  
+  "body": {
+    "id": "1",
+    "severity": "debug",
+    "message": "\"Debug Message\""
+  }
+}
+```
+
+</td>
+</tr>
+</table>
+
+#### Parse the field `message` with csv parser and also parse the timestamp
+
+Configuration:
+
+```yaml
+- type: csv_parser
+  header: 'timestamp_field,severity,message'
+  timestamp:
+    parse_from: timestamp_field
+    layout_type: strptime
+    layout: '%Y-%m-%d'
+```
+
+<table>
+<tr><td> Input Entry </td> <td> Output Entry </td></tr>
+<tr>
+<td>
+
+```json
+{
+  "timestamp": "",
+  "resource": { },
+  "attributes": { },  
```

Comment:
why do we want to remove these? I thought we wanted to have the entry standardized throughout the different examples and ops. like you said for the flatten op here https://github.com/open-telemetry/opentelemetry-log-collection/pull/77#discussion_r616104818

Reply:
To clarify, we should use a standard format, but only show the fields that are relevant to the example.

For instance, we don't typically show `"severity"` unless it is relevant. 

The general idea is to avoid distracting from the point of the example.